backend/src/apiserver/resource/model_converter.go

```diff
@@ -244,6 +246,8 @@ func (r *ResourceManager) getResourceName(resourceType common.ResourceType, reso
 			return "", util.Wrap(err, "Referred pipeline version not found.")
 		}
 		return version.Name, nil
+	case common.Namespace:
```

Comment:
What would happen if user delete the namespace and later someone else create a namespace with the same name? Would it cause the data leak?
Should we store the uuid of a namespace instead? (this is not super urgent but we might want to think through)

Reply:
That is a very good point. But, could it be that reusing the same name is the intended behavior? I will put a note here for further investigation.